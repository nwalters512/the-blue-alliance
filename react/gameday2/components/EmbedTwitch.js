import React, { PropTypes } from 'react';

var EmbedTwitch = React.createClass({
  propTypes: {
    webcast: PropTypes.object.isRequired
  },
  render: function() {
    var channel = this.props.webcast.channel;
    var src = `http://www.twitch.tv/widgets/live_embed_player.swf?channel=${channel}`;
    var flashvars = `hostname=www.twitch.tv&channel=${channel}&auto_play=true&start_volume=25&enable_javascript=true`;
    return (
      <object
        type='application/x-shockwave-flash'
        height='100%'
        width='100%'
        id='live_embed_player_flash'
        data={src}
        bgcolor='#000000'>
        <param name='allowFullScreen' value='true' />
        <param name='allowScriptAccess' value='always' />
        <param name='allowNetworking' value='all' />
        <param
          name='movie'
          value='http://www.twitch.tv/widgets/live_embed_player.swf' />
        <param
          name='flashvars'
          value={flashvars} />
        <param name='wmode' value='transparent' />
      </object>
    );
  }
});

export default EmbedTwitch;
