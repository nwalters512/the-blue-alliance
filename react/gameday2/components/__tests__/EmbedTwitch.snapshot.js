import React from 'react'
import renderer from 'react-test-renderer'
import EmbedTwitch from '../EmbedTwitch'

describe('EmbedTwitch snapshot', () => {
  it('renders correctly', () => {
    const webcast = {
      channel: 'test',
    }

    const component = renderer.create(
      <EmbedTwitch webcast={webcast} />
    ).toJSON()
    expect(component).toMatchSnapshot()
  })
})
