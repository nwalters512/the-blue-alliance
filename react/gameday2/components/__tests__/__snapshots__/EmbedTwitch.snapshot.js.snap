exports[`EmbedTwitch snapshot renders correctly 1`] = `
<iframe
  frameBorder="0"
  height="100%"
  scrolling="no"
  src="https://player.twitch.tv/?channel=test"
  width="100%" />
`;
